pkg/utils/utils_virtual.go

```diff
@@ -205,6 +227,33 @@ func getOpenstackDataFromMetadataService() (metaData *OSPMetaData, networkData *
 	return metaData, networkData, nil
 }
 
+// getPCIAddressFromMACAddress returns the PCI address of a device given its MAC address
+func getPCIAddressFromMACAddress(macAddress string) (string, error) {
+	nics, err := ghw.Network()
+	if err != nil {
+		return "", fmt.Errorf("error getting network info: %w", err)
+	}
+
+	var pciAddress string
+	for _, nic := range nics.NICs {
+		if strings.EqualFold(nic.MacAddress, macAddress) {
+			if pciAddress == "" {
+				pciAddress = *nic.PCIAddress
+			} else {
+				// Check if there are more than one device with the same MAC address
+				// and return an error if that's the case, we don't support that scenario for now.
```

Comment:
comment is not needed IMO. i think the error below is clear. feel free to throw in  "unsupported" somewhere below if you  you think its worth it.